### PR TITLE
Update sui-glossary.md

### PR DESCRIPTION
**Pull request by Clay-Mysten**
Add Objects to Sui Glossary and summarize potential states

### CODE REVIEW

File: doc/src/learn/sui-glossary.md [hunk 1]
```diff
@@ -79,6 +79,21 @@ exchange of private keys, users can offload the bandwidth use of transaction sub
...
+
+* *Immutable* - the object cannot be modified.
+* *Mutable* - the object can be changed.
+
+Further, mutable objects are divided into these categories:
+
+* *Owned* - the object can be modified only by its owner.
+* *Shared* - the object can be modified by anyone.
+
+Immutable objects are always considered shared.
```
Comment (tnowacki) [inline doc/src/learn/sui-glossary.md:94]:
```suggestion
Immutable objects are always considered "shared", in the sense that they do not have an owner.
```

Trying really hard not to directly link the two, avoiding the shared v shared immutable v shared mutable situation from before. But I think it is helpful in a non-formal way for conveying meaning to talk about immutable objects not being owned by anyone.

Comment (lxfind) [reply doc/src/learn/sui-glossary.md:94]: Agreed. I would prefer we even remove "shared" here to avoid confusions.

**Comment by Clay-Mysten** [comment]
Xun, let me know when this content reflects the latest terminology, and I will re-review. Thanks!

**Review by lxfind — APPROVED** [review]
LGTM. We can merge this now

**Pull request merged**